api/proto/teleport/legacy/types/types.proto

```diff
@@ -6812,6 +6812,10 @@ message PluginAWSICSettings {
   // SAMLIdPServiceProviderName is the name of a SAML service provider created
   // for the Identity Center.
   string saml_idp_service_provider_name = 6;
+
+  // UseAmbientAwsCreds indicates that the integration should use the ambient
+  // AWS credentials rather than
+  bool use_ambient_aws_creds = 7;
```

Comment:
Can we use an existing method to distinguish credentials?

Can we consider reusing the approach introduced in the Entra ID integration, as defined in the following file:
[Link to code](https://github.com/gravitational/teleport/blob/master/api/proto/teleport/legacy/types/types.proto#L6733)

In this integration, credentials are distinguished by SOURCE_OIDC and SOURCE_SYSTEM_CREDENTIALS.

WDYT ?

Reply:
Yeah, makes sense the number of times we've had to shift from a boolean to an enum when other options turn up.